spark/src/test/scala/org/apache/comet/CometExpressionSuite.scala

```diff
@@ -1469,5 +1469,38 @@ class CometExpressionSuite extends CometTestBase with AdaptiveSparkPlanHelper {
       }
     }
   }
+  test("unary negative integer overflow test") {
 
+    val df = Seq(Int.MaxValue, Int.MinValue).toDF("a")
+    df.write.mode("overwrite").parquet("/tmp/int.parquet")
+    spark.read.parquet("/tmp/int.parquet").createTempView("t")
+
+    // without ANSI mode
+    withSQLConf(
+      SQLConf.ANSI_ENABLED.key -> "false",
+      CometConf.COMET_ANSI_MODE_ENABLED.key -> "false",
+      CometConf.COMET_ENABLED.key -> "true",
+      CometConf.COMET_EXEC_ENABLED.key -> "true") {
+      checkSparkMaybeThrows(sql("select a, -a from t")) match {
+        case (None, None) => // no exception
+        case _ =>
+          fail("No exception should be thrown")
+      }
```

Comment:
I think you can just use `checkSparkAnswerAndOperator` here to make sure Comet returns the correct results and does not throw an exception.

```suggestion
      checkSparkAnswerAndOperator(sql("select a, -a from t"))
```